src/theory/arith/nl/ext/monomial_bounds_check.cpp

```diff
@@ -294,19 +294,30 @@ void MonomialBoundsCheck::checkBounds(const std::vector<Node>& asserts,
                 << "     ...coefficient " << mult << " is zero." << std::endl;
             continue;
           }
+          Node lhsTgt = t;
+          Node rhsTgt = rhs;
+          // if we are making an equality below, we require making it
+          // well-typed so that lhs/rhs have the same type. We use the
+          // mkEquality utility to do this
+          if (type == kind::EQUAL)
+          {
+            Node eq = mkEquality(lhsTgt, rhsTgt);
```

Comment:
This seems a bit weird. Maybe we should introduce a new utility like this:
```
std::pair<Node,Node> mkSameType(const Node& a, const Node& b) {
  if (a.getType() == b.getType()) return {a, b};
  if (a.isInt() && b.isReal()) return {TO_REAL(a), b};
  Assert(a.isReal() && b.isInt());
  return {a, TO_REAL(b)};
}

std::tie(lhsTgt, rhsTgt) = mkSameType(lhsTgt, rhsTgt);
```

Reply:
Ok, I'll add this.